packages/validator/contracts/ERC20Executor.sol

```diff
@@ -0,0 +1,72 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.10;
+
+import "./SingleClaimantExecutor.sol";
+import "@openzeppelin/contracts/contracts/token/ERC20/IERC20.sol";
+
+/**
+ * @dev An abstract implementation of an Executor that only allows claims that have a higher nonce than previously seen to execute.
+ *      To use, subclass and override `executeClaim` and `metadata`, being sure to call `super` inside `executeClaim` before doing anything else.
+ *      This executor expects the first 32 bytes of `claimData` to be the nonce; you may optionally use extra data for your own purposes.
+ */
+abstract contract ERC20Executor is SingleClaimantExecutor {
```

Comment:
How about making this just inherit `BaseExecutor`, so people can mix-and-match replay-protection strategy and distribution method?